### DISCUSSION TITLE
Cooldown time or ignoring failing PRs

### DISCUSSION THREAD

**Discussion by Sebazzz** [renovatebot/renovate#25907] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

HEAD

### If you're self-hosting Renovate, select which platform you are using.

Gitea or Forgejo

### What is your question?

In my experience when a library is upgraded and the PR automated tests fail, because of some incompatible changes, Renovate keeps rebasing and retrying the PR.

Is it possible for Renovate to, at some point, ignore the PR and try and update other dependencies, and perhaps retry the PR after?

### Logs (if relevant)

<details><summary>Logs</summary>

```

Copy/paste the relevant log(s) here, between the starting and ending backticks

```

</details>

**Comment by rarkins** [comment]
This feature would help users manually stop failing PRs: #20940

If you want something more automated, you'll need to suggest something more specific. For example "don't rebase if the current PR has failing tests". This would need to be opt-in, because it would be undesirable default behavior for many

**Reply by Sebazzz** [reply]
It work be basically something like this:

- Retry the PR a maximum of X times
- If the PR fails, ignore it for a while. Don't count this PR in the maximum number of open PRs.
- After a certain cooldown time, allow retrying the PR again.

This makes sure other dependencies can still be updated by Renovate, even if there are a few that cannot be updated automatically due to breaking changes. Currently it seems Renovate is restricted and will be stalled by these few dependencies that are failing.

**Reply by rarkins** [reply]
I disagree completely with the middle point about not counting the PR

**Reply by Sebazzz** [reply]
Well in that case you still have the issue that Renovate won't try any other PR (or you'd have to close them automatically).

**Reply by rarkins** [reply]
Yes, so you can either:
- Close the PR to ignore it, opening up space for others, or
- Use the Dependency Dashboard issue to manually approve others to open

**Reply by Sebazzz** [reply]
Currently yes. Is there any merit to my case to ask this as a feature request?

**Reply by rarkins** [reply]
I think that the a "maximum X rebases if tests keep failing" has merits and could be implemented first, essentially with an infinite cooling-off period. Configurable cooling-off is something I'd consider low priority and potentially not worth the complexity
